### PR TITLE
Move `Catalyst`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
Motivation
Progress https://github.com/optuna/optuna/issues/4484

Description of the changes
- Move `integration/catalyst.py` and its test from `optuna/optuna`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ integrated with many useful tools like PyTorch, sklearn, TensorFlow, etc.
 Optuna-Integration API reference is [here](https://optuna.readthedocs.io/en/stable/reference/integration.html).
 
 * [AllenNLP](https://optuna.readthedocs.io/en/stable/reference/integration.html#allennlp)  ([example](https://github.com/optuna/optuna-examples/tree/main/allennlp))
+* [Catalyst](https://optuna.readthedocs.io/en/stable/reference/integration.html#catalyst)
```
Comment (HideakiImamura) [inline README.md:19]: Please point out the optuna-integration's side document.

File: optuna_integration/catalyst.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from optuna._deprecated import deprecated_class
+from optuna._imports import try_import
```
Comment (HideakiImamura) [inline optuna_integration/catalyst.py:2]: After #16, please use optuna-integration's side `_imports`.

File: tests/test_catalyst.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+
+pytestmark = pytest.mark.integration
+
+
+def test_warning() -> None:
+    study = optuna.create_study()
+    trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
+    with pytest.warns(FutureWarning):
+        optuna.integration.CatalystPruningCallback(
```
Comment (HideakiImamura) [inline tests/test_catalyst.py:12]: Please test optuna_integration's side function.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have several comments. PTAL.

**Comment by Alnusjaponica** [comment]
@HideakiImamura Thanks for your review. I fixed all of them

File: README.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ integrated with many useful tools like PyTorch, sklearn, TensorFlow, etc.
 Optuna-Integration API reference is [here](https://optuna-integration.readthedocs.io/en/stable/reference/index.html).
 
 * [AllenNLP](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#allennlp)  ([example](https://github.com/optuna/optuna-examples/tree/main/allennlp))
+* [Catalyst](https://optuna-integration.readthedocs.io/en/stable/reference/index.html#catalyst)
```
Comment (HideakiImamura) [inline README.md:21]: Please add a link to https://github.com/optuna/optuna-examples/blob/main/pytorch/catalyst_simple.py.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the update. I have a comment. PTAL.

File: README.md (hunk 4 above)
Comment (Alnusjaponica) [reply README.md]: Thanks for the comment. I updated `README.md`.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**